lib/server/openload.py

```diff
@@ -179,98 +31,24 @@ def getNumberFromString(self, sValue):
-        new_num_string = remainder_string + new_num_string
-        current = current / n
-    return new_num_string
 
 def resolve(url):
+    video_code = url.split('/')[-2]
+    api_url = 'https://api.openload.co/1/streaming/get?file=%s' % video_code
+    print('api_url: %s' % api_url)
 
-    oRequest = cRequestHandler(url)
-    sHtmlContent = oRequest.request()
-
-    oParser = cParser()
-    string = ''
-
-    sPattern = '<script type="text\/javascript">(ﾟωﾟ.+?)<\/script>'
-    aResult = oParser.parse(sHtmlContent, sPattern)
-
-    vid = 'XXXXXX'
-    string2=[]
-
-    for aEntry in aResult[1]:
-        s = AADecoder(aEntry).decode()
-        string2.append(s)
-
-        if 'welikekodi_ya_rly' in s:
-            c0 = re.search('welikekodi_ya_rly = ([^<>;"]+);', s)
-            if c0:
-                c = c0.group(1)
-                c = c.replace('Math.round','int')
-                cc = str(eval(c))
-                vid = '[' + cc + ']'
-
-    for string3 in string2:
-        if ('toString' in string3) and (vid in string3):
-            base = int(re.findall('toString\(a\+([0-9]+)\)',string3)[0])
-            table = re.findall('(\([0-9][^)]+\))',string3)
-
-            for str1 in table:
-                val = re.findall('([0-9]+),([0-9]+)',str1)
-                base2 = base + int(val[0][0])
-                str2 = base10toN(int(val[0][1]), base2)
-                string3 = string3.replace(str1, str2)
-
-            #clean up
-            string3 = string3.replace('+', '')
-            string3 = string3.replace('"', '')
-            string3 = string3.replace('', '')
-
-            #a hack for not having to recode everything
-            url = re.findall('(http[^<>}]+)',string3)[0]
-            string = 'src="' + url + '?mime=true"'
-
-    if (string):
-        sContent = string.replace('\\','')
-
-        api_call = ''
-
-        sPattern = 'src=\s*?"(.*?)\?'
-        aResult = oParser.parse(sContent, sPattern)
-
-        if (aResult[0] == True):
-            api_call = aResult[1][0]
-
-        if not api_call:
-            sPattern = 'window\.vr *=["\'](.+?)["\']'
-            aResult = oParser.parse(sContent, sPattern)
-            if (aResult[0] == True):
-                api_call = aResult[1][0]
-
-    if (api_call):
-
-        if 'openload.co/stream' in api_call:
-            UA = 'Mozilla/5.0 (Windows NT 6.1; WOW64; rv:39.0) Gecko/20100101 Firefox/39.0'
-            headers = {'User-Agent': UA }
+    request = urllib2.Request(api_url)
 
-            req = urllib2.Request(api_call,None,headers)
-            res = urllib2.urlopen(req)
-            finalurl = res.geturl()
-            api_call = finalurl
+    page = urllib2.urlopen(request)
+    sHtmlContent = page.read()
+    page.close()
 
-    return [{'url': api_call}]
+    response = demjson.decode(sHtmlContent)
+    print(response['result'])
+    if response['msg'] not in 'OK':
+        xbmc.executebuiltin('Notification(%s, %s)' % ('Openload resolver',
```

Comment:
@lzoubek Doesn't like using `xbmc` module in resolvers because of portability issues. Could we work this around without using `xbmc`? Anyway, why can't we do this for the user automatically?